tests/unit/s2n_session_ticket_test.c

```diff
@@ -1348,6 +1348,54 @@ int main(int argc, char **argv)
 
+    /* Test: TLS1.3 resumption is successful when key used to encrypt ticket is in decrypt-only state */
+    if (s2n_is_tls13_fully_supported()) {
+        /* Initialize client and server connections with TLS 1.3 configurations */
+        DEFER_CLEANUP(struct s2n_connection *client_connection = s2n_connection_new(S2N_CLIENT), s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(client_connection);
+
+        EXPECT_SUCCESS(s2n_connection_set_session(client_connection, tls13_serialized_session_state.blob.data,
+                s2n_stuffer_data_available(&tls13_serialized_session_state)));
+
+        DEFER_CLEANUP(struct s2n_config *client_configuration = s2n_config_new(), s2n_config_ptr_free);
+        EXPECT_NOT_NULL(client_configuration);
+        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(client_configuration, 1));
+        EXPECT_SUCCESS(s2n_connection_set_config(client_connection, client_configuration));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(client_configuration, "default_tls13"));
+
+        DEFER_CLEANUP(struct s2n_connection *server_connection = s2n_connection_new(S2N_SERVER), s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(server_connection);
+
+        DEFER_CLEANUP(struct s2n_config *server_configuration = s2n_config_new(), s2n_config_ptr_free);
+        EXPECT_NOT_NULL(server_configuration);
+        EXPECT_SUCCESS(s2n_config_set_session_tickets_onoff(server_configuration, 1));
+        EXPECT_SUCCESS(s2n_config_set_cipher_preferences(server_configuration, "default_tls13"));
+
+        /* Add the key that encrypted the session ticket so that the server will be able to decrypt the ticket successfully. */
+        EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(server_configuration, ticket_key_name1,
+                s2n_array_len(ticket_key_name1), ticket_key1, s2n_array_len(ticket_key1), 0));
+
+        /* Add a mock delay such that key 1 moves to decrypt-only state */
+        uint64_t mock_delay = server_configuration->encrypt_decrypt_key_lifetime_in_nanos;
+        EXPECT_SUCCESS(s2n_config_set_wall_clock(server_configuration, mock_nanoseconds_since_epoch, &mock_delay));
+
+        /* Add one session ticket key with an intro time in the past so that the key is immediately valid */
+        POSIX_GUARD(server_configuration->wall_clock(server_configuration->sys_clock_ctx, &now));
+        uint64_t key_intro_time = (now / ONE_SEC_IN_NANOS) - ONE_SEC_DELAY;
+        EXPECT_SUCCESS(s2n_config_add_ticket_crypto_key(server_configuration, ticket_key_name2,
+                s2n_array_len(ticket_key_name2), ticket_key2, s2n_array_len(ticket_key2), key_intro_time));
+
+        EXPECT_SUCCESS(s2n_connection_set_config(server_connection, server_configuration));
+
+        /* Create nonblocking pipes */
+        EXPECT_SUCCESS(s2n_connections_set_io_pair(client_connection, server_connection, &io_pair));
+
+        EXPECT_SUCCESS(s2n_negotiate_test_server_and_client(server_connection, client_connection));
+
```

Comment:
If this test relies on TLS1.3, you should probably assert that it negotiated TLS1.3.